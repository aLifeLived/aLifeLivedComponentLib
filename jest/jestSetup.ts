/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
