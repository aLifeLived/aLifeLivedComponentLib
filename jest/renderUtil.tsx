import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
