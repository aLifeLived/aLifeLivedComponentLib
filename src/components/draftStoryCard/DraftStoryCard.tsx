import React from 'react';
import ProgressBar from 'react-native-animated-progress';

// UI
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';
import { SystemButton } from '../../themes/systemButton/SystemButton';
import { Icon } from '../icon/Icon';
import { useTheme } from '../../themes/hooks/useTheme';
import { SystemPressable } from '../../themes/systemPressable/SystemPressable';

type DraftStoryCardProps = {
  title: string;
  topicTitle: string;
  recordingProgress: number;
  shouldDisplayEllipsis?: boolean;
  onPress: () => void;
  onEllipsisPress?: () => void;
};

export const DraftStoryCard: React.FC<DraftStoryCardProps> = ({
  title,
  topicTitle,
  recordingProgress,
  shouldDisplayEllipsis,
  onPress,
  onEllipsisPress,
}) => {
  const theme = useTheme();
  return (
    <SystemButton onPress={onPress}>
      <Box
        borderColor="borderColor"
        backgroundColor="backgroundContrastXLow"
        borderWidth={1}
        borderRadius="medium"
        flexDirection="row"
        alignContent="center"
        alignItems="center"
        justifyContent="space-between"
        boxShadow="sm"
        paddingBottom="s"
        overflow="hidden"
      >
        <Box
          marginLeft="s"
          marginBottom="s"
          flexDirection="row"
          paddingTop="s"
          alignItems="center"
        >
          <Box
            padding="s"
            marginRight="m"
            backgroundColor="backgroundContrastLow"
            borderRadius="large"
            testID="pencil-icon"
          >
            <Icon icon="pencil-alt" color="primary" />
          </Box>
          <Box>
            <Text color="textContrastHigh" variant="intro" numberOfLines={1}>
              {topicTitle}
            </Text>

            <Text
              variant="bodyXSmall"
              color="textContrastHigh"
              numberOfLines={1}
            >
              {title}
            </Text>
          </Box>
        </Box>
        {shouldDisplayEllipsis ? (
          <SystemPressable
            onPress={onEllipsisPress}
            paddingTop="m"
            paddingBottom="m"
            accessibilityLabel="Draft story more options"
          >
            <Icon
              icon="ellipsis-v"
              color="primary"
              size={20}
              marginRight="m"
              testID="ellipsis-icon"
            />
          </SystemPressable>
        ) : null}
        <Box
          width="100%"
          position="absolute"
          bottom={0}
          backgroundColor="textError"
        >
          <ProgressBar
            progress={recordingProgress}
            backgroundColor={theme.colors.primary}
            trackColor={theme.colors.backgroundContrastMid}
            height={3.5}
          />
        </Box>
      </Box>
    </SystemButton>
  );
};
