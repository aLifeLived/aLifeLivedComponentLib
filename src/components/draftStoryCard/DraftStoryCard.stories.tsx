import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { text, number } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

// Component
import { DraftStoryCard } from './DraftStoryCard';
import { Box } from '../../themes/box/Box';

storiesOf('Draft story card', module)
  .addDecorator(Story => Story())
  .add('Default draft story card', () => (
    <Box margin="m">
      <DraftStoryCard
        title={text('Title', 'This is a test title')}
        topicTitle={text('Topic Title', 'Life story')}
        onPress={action('On press fired')}
        storyProgress={number('Story progress', 10)}
      />
    </Box>
  ));
