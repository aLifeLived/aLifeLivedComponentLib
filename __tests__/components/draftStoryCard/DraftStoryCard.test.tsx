import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { DraftStoryCard } from '../../../src/components/draftStoryCard/DraftStoryCard';

const onPress = jest.fn();
const onEllipsisPress = jest.fn();
const defaultTitle = 'Test Title';
const defaultTopicTitle = 'This is a test topic';

const setup = (storyProgress = 0) =>
  render(
    <DraftStoryCard
      onEllipsisPress={onEllipsisPress}
      recordingProgress={storyProgress}
      title={defaultTitle}
      topicTitle={defaultTopicTitle}
      onPress={onPress}
    />
  );

describe('components >> draft story card ', () => {
  beforeEach(() => onPress.mockClear());
  test('Should match snapshot correctly', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should find title and topic text correctly', () => {
    const { getByText } = setup();
    expect(getByText(defaultTitle)).toBeDefined();
    expect(getByText(defaultTopicTitle)).toBeDefined();
  });

  test('Should fire on press action', () => {
    const { getByText } = setup();
    fireEvent.press(getByText(defaultTitle));
    expect(onPress).toBeCalledTimes(1);
  });

  test('Should find pencil icon', () => {
    const { getByTestId } = setup();
    expect(getByTestId('pencil-icon')).toBeDefined();
  });

  describe('Given user wants to delete draft', () => {
    test('Should fire onEllipsisPress', () => {
      const { getByTestId } = setup();
      fireEvent.press(getByTestId('ellipsis-icon'));
      expect(onEllipsisPress).toBeCalledTimes(1);
    });
  });

  describe('Given user has progressed through 50% of recording', () => {
    test('Should render progress bar of half way', () => {
      const { toJSON } = setup(50);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
