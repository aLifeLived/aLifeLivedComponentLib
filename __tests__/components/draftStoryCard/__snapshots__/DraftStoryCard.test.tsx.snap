// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> draft story card  Should match snapshot correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#252424",
          "borderRadius": 8,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 16,
        },
        Array [
          Object {
            "elevation": 2,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 10,
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "marginBottom": 8,
            "marginLeft": 8,
            "width": "80%",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#252424",
              "display": "flex",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
              "marginRight": 16,
              "marginTop": 8,
            },
          ]
        }
      >
        Test Title
      </Text>
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#0FCFA6",
              "borderRadius": 8,
              "display": "flex",
              "marginTop": 8,
              "padding": 2,
              "paddingLeft": 8,
              "paddingRight": 8,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "display": "flex",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          This is a test topic
        </Text>
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="#252424"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      size={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "marginRight": 16,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={512}
      vbWidth={320}
      width={20}
    >
      <RNSVGGroup
        fill={4280624164}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill={4280624164}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
